stripe_test.go

```diff
@@ -149,142 +149,188 @@ func TestShouldRetry(t *testing.T) {
 	).(*BackendImplementation)
 
 	// Exceeded maximum number of retries
-	assert.False(t, c.shouldRetry(
-		nil,
-		&http.Request{},
-		&http.Response{},
-		int(MaxNetworkRetries),
-	))
+	t.Run("DontRetryOnExceededRetries", func(t *testing.T) {
```

Comment:
I also refactored this test case to use subtests, which is a pretty typical pattern in Go. This has a few advantages:

* Subtests run inside their own closure `func ... { ... }` so variables are localized to them, which means variables can't leak into other subtests and keeps things generally cleaner.
* Go tooling allows us to run any specific subtest with `-name` so we can target exactly one of them without having to run every single one.